cpp/benchmarks/fixture/nvbench_fixture.hpp

```diff
@@ -90,12 +92,12 @@ struct nvbench_base_fixture {
 
   inline rmm::host_async_resource_ref make_cuio_host_pinned_pool()
   {
-    using host_pooled_mr = rmm::mr::pool_memory_resource<rmm::mr::pinned_host_memory_resource>;
-    static std::shared_ptr<host_pooled_mr> mr = std::make_shared<host_pooled_mr>(
+    // Don't store in static, as the CUDA context may be destroyed before static destruction
+    this->host_pooled_mr = std::make_shared<host_pooled_mr_t>(
```

Comment:
Should this function guard against being called twice?

Reply:
Yes -- fixed.